Fix prometheus service

teochenglim opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

```shell
$ brew install --build-from-source --debug --verbose Formula/prometheus.rb
Updating Homebrew...
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FromPathLoader): loading /Users/teochenglim/homebrew-core/Formula/prometheus.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/go.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/prometheus.rb
/usr/bin/sandbox-exec -f /private/tmp/homebrew20191111-68318-102r137.sb nice ruby -W0 -I $LOAD_PATH -- /usr/local/Homebrew/Library/Homebrew/build.rb /Users/teochenglim/homebrew-core/Formula/prometheus.rb --verbose --debug
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FromPathLoader): loading /Users/teochenglim/homebrew-core/Formula/prometheus.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/go.rb
==> Downloading https://github.com/prometheus/prometheus/archive/v2.13.1.tar.gz
/usr/bin/curl -q --globoff --show-error --user-agent Homebrew/2.1.16-38-ge2c76cc\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 10.15.1\)\ curl/7.64.1 --location --silent --head --request GET https://github.com/prometheus/prometheus/archive/v2.13.1.tar.gz
Already downloaded: /Users/teochenglim/Library/Caches/Homebrew/downloads/3b968a8dc1e632bfaa45e1d701b30be5c621530090ba398074d0ee5e83169ad9--prometheus-2.13.1.tar.gz
==> Verifying 3b968a8dc1e632bfaa45e1d701b30be5c621530090ba398074d0ee5e83169ad9--prometheus-2.13.1.tar.gz checksum
tar xof /Users/teochenglim/Library/Caches/Homebrew/downloads/3b968a8dc1e632bfaa45e1d701b30be5c621530090ba398074d0ee5e83169ad9--prometheus-2.13.1.tar.gz -C /private/tmp/d20191111-68319-ajh6km
cp -pR /private/tmp/d20191111-68319-ajh6km/prometheus-2.13.1/. /private/tmp/prometheus-20191111-68319-1ktt95u/prometheus-2.13.1
chmod -Rf +w /private/tmp/d20191111-68319-ajh6km
==> make build
curl -s -L https://github.com/prometheus/promu/releases/download/v0.5.0/promu-0.5.0.darwin-amd64.tar.gz | tar -xvzf - -C /var/folders/dp/qtmghhxn2d7d4xfcfk0qwl300000gp/T/tmp.lihovtj2
x promu-0.5.0.darwin-amd64/
x promu-0.5.0.darwin-amd64/promu
x promu-0.5.0.darwin-amd64/NOTICE
x promu-0.5.0.darwin-amd64/LICENSE
mkdir -p /private/tmp/prometheus-20191111-68319-1ktt95u/prometheus-2.13.1/.brew_home/go/bin
cp /var/folders/dp/qtmghhxn2d7d4xfcfk0qwl300000gp/T/tmp.lihovtj2/promu-0.5.0.darwin-amd64/promu /private/tmp/prometheus-20191111-68319-1ktt95u/prometheus-2.13.1/.brew_home/go/bin/promu
rm -r /var/folders/dp/qtmghhxn2d7d4xfcfk0qwl300000gp/T/tmp.lihovtj2
>> building binaries
GO111MODULE=on /private/tmp/prometheus-20191111-68319-1ktt95u/prometheus-2.13.1/.brew_home/go/bin/promu build --prefix /private/tmp/prometheus-20191111-68319-1ktt95u/prometheus-2.13.1
 >   prometheus
 >   promtool
 >   tsdb
==> Cleaning
Fixing /usr/local/Cellar/prometheus/2.13.1/bin/prometheus permissions from 755 to 555
Fixing /usr/local/Cellar/prometheus/2.13.1/bin/promtool permissions from 755 to 555
==> Finishing up
ln -s ../Cellar/prometheus/2.13.1/bin/prometheus prometheus
ln -s ../Cellar/prometheus/2.13.1/bin/promtool promtool
/usr/bin/sandbox-exec -f /private/tmp/homebrew20191111-70325-1sey3vu.sb nice ruby -W0 -I $LOAD_PATH -- /usr/local/Homebrew/Library/Homebrew/postinstall.rb /Users/teochenglim/homebrew-core/Formula/prometheus.rb --build-from-source --debug --verbose
/usr/local/Homebrew/Library/Homebrew/postinstall.rb (Formulary::FromPathLoader): loading /Users/teochenglim/homebrew-core/Formula/prometheus.rb
==> echo '--config.file $(brew --prefix)/etc/prometheus.yml        --web.listen-address=127.0.0.1:9090       --storage.tsdb.path $(brew --prefix)/var/prometheus'       > $(brew --prefix)/etc/prometheus.args
==> Caveats
When used with `brew services`, prometheus' configuration is stored as command line flags in
  /usr/local/etc/prometheus.args

Example configuration:
  echo '--config.file $(brew --prefix)/etc/prometheus.yml    --web.listen-address=127.0.0.1:9090   --storage.tsdb.path $(brew --prefix)/var/prometheus'   > $(brew --prefix)/etc/prometheus.args


To have launchd start prometheus now and restart at login:
  brew services start prometheus
Or, if you don't want/need a background service you can just run:
  prometheus
==> Summary
🍺  /usr/local/Cellar/prometheus/2.13.1: 18 files, 119.3MB, built in 4 minutes 39 seconds



$ brew test prometheus
Testing prometheus
==> /usr/local/Cellar/prometheus/2.13.1/bin/promtool check rules /private/tmp/prometheus-test-20191111-70716-14gfvb9/rules.example


$ brew audit --strict prometheus
Fetching gem metadata from https://rubygems.org/.........
Fetching concurrent-ruby 1.1.5
Fetching thread_safe 0.3.6
Fetching minitest 5.13.0
Installing thread_safe 0.3.6
Installing minitest 5.13.0
Installing concurrent-ruby 1.1.5
Fetching zeitwerk 2.2.1
Installing zeitwerk 2.2.1
Fetching ast 2.4.0
Installing ast 2.4.0
Using bundler 1.17.2
Fetching connection_pool 2.2.2
Installing connection_pool 2.2.2
Fetching json 2.2.0
Fetching docile 1.3.2
Installing json 2.2.0 with native extensions
Installing docile 1.3.2
Fetching simplecov-html 0.10.2
Installing simplecov-html 0.10.2
Fetching tins 1.22.0
Installing tins 1.22.0
Fetching thor 0.20.3
Installing thor 0.20.3
Fetching diff-lcs 1.3
Installing diff-lcs 1.3
Fetching unf_ext 0.0.7.6
Installing unf_ext 0.0.7.6 with native extensions
Fetching hpricot 0.8.6
Installing hpricot 0.8.6 with native extensions
Fetching jaro_winkler 1.5.4
Installing jaro_winkler 1.5.4 with native extensions
Fetching mime-types-data 3.2019.1009
Installing mime-types-data 3.2019.1009
Fetching net-http-digest_auth 1.4.1
Installing net-http-digest_auth 1.4.1
Fetching mini_portile2 2.4.0
Fetching ntlm-http 0.1.1
Installing mini_portile2 2.4.0
Installing ntlm-http 0.1.1
Fetching webrobots 0.1.2
Installing webrobots 0.1.2
Fetching mustache 1.1.0
Fetching parallel 1.18.0
Installing mustache 1.1.0
Installing parallel 1.18.0
Fetching plist 3.5.0
Installing plist 3.5.0
Fetching rainbow 3.0.0
Installing rainbow 3.0.0
Fetching rdiscount 2.2.0.1
Fetching rspec-support 3.9.0
Installing rdiscount 2.2.0.1 with native extensions
Installing rspec-support 3.9.0
Fetching ruby-progressbar 1.10.1
Installing ruby-progressbar 1.10.1
Fetching unicode-display_width 1.6.0
Installing unicode-display_width 1.6.0
Fetching ruby-macho 2.2.0
Installing ruby-macho 2.2.0
Fetching tzinfo 1.2.5
Installing tzinfo 1.2.5
Fetching parser 2.6.5.0
Installing parser 2.6.5.0
Fetching net-http-persistent 3.1.0
Installing net-http-persistent 3.1.0
Fetching i18n 1.7.0
Installing i18n 1.7.0
Fetching term-ansicolor 1.7.1
Installing term-ansicolor 1.7.1
Fetching unf 0.1.4
Installing unf 0.1.4
Fetching mime-types 3.3
Installing mime-types 3.3
Fetching simplecov 0.16.1
Installing simplecov 0.16.1
Fetching nokogiri 1.10.5
Installing nokogiri 1.10.5 with native extensions
Fetching parallel_tests 2.29.2
Installing parallel_tests 2.29.2
Fetching rspec-core 3.9.0
Installing rspec-core 3.9.0
Fetching rspec-expectations 3.9.0
Installing rspec-expectations 3.9.0
Fetching rspec-mocks 3.9.0
Installing rspec-mocks 3.9.0
Fetching rubocop 0.76.0
Installing rubocop 0.76.0
Fetching activesupport 6.0.1
Installing activesupport 6.0.1
Fetching domain_name 0.5.20190701
Installing domain_name 0.5.20190701
Fetching coveralls 0.8.23
Installing coveralls 0.8.23
Fetching rspec-retry 0.6.1
Installing rspec-retry 0.6.1
Fetching rspec-its 1.3.0
Installing rspec-its 1.3.0
Fetching rspec 3.9.0
Installing rspec 3.9.0
Fetching rubocop-performance 1.5.0
Installing rubocop-performance 1.5.0
Fetching rubocop-rspec 1.36.0
Installing rubocop-rspec 1.36.0
Fetching http-cookie 1.0.3
Installing http-cookie 1.0.3
Fetching rspec-wait 0.0.9
Fetching ronn 0.7.3
Installing rspec-wait 0.0.9
Installing ronn 0.7.3
Fetching mechanize 2.7.6
Installing mechanize 2.7.6
Bundle complete! 16 Gemfile dependencies, 56 gems now installed.
Bundled gems are installed into `../../../usr/local/Homebrew/Library/Homebrew/vendor/bundle`
Post-install message from i18n:

HEADS UP! i18n 1.1 changed fallbacks to exclude default locale.
But that may break your application.

Please check your Rails app for 'config.i18n.fallbacks = true'.
If you're using I18n (>= 1.1.0) and Rails (< 5.2.2), this should be
'config.i18n.fallbacks = [I18n.default_locale]'.
If not, fallbacks will be broken in your app by I18n 1.1.x.

For more info see:
https://github.com/svenfuchs/i18n/releases/tag/v1.1.0

```

SMillerDev commented on Formula/prometheus.rb:
```diff
@@ -27,7 +27,10 @@ def caveats; <<~EOS
 
     Example configuration:
-      echo "--config.file ~/.config/prometheus.yml" > #{etc}/prometheus.args
+      echo '--config.file $(brew --prefix)/etc/prometheus.yml  \
```
This shouldn't be in the caveat if it's already done.

teochenglim replied:
Agreed. Removed lines now. and submitted the changes.

SMillerDev commented:
This should resolve #44508

teochenglim commented:
Hi @SMillerDev, added prometheus.yml file by default also. So that user can just "brew services start prometheus" now after installation. 

Prometheus default config get from official documentation.
https://prometheus.io/docs/introduction/first_steps/

SMillerDev commented on Formula/prometheus.rb:
```diff
@@ -22,12 +22,20 @@ def install
+    system "echo '--config.file #{etc}/prometheus.yml  \
+      --web.listen-address=127.0.0.1:9090 \
+      --storage.tsdb.path #{var}/prometheus' \
+      > #{etc}/prometheus.args"
```
This might be a better way
```ruby
    (etc/"prometheus.args").write <<~EOS
      --config.file #{etc}/prometheus.yml
      --web.listen-address=127.0.0.1:9090
      --storage.tsdb.path #{var}/prometheus
    EOS
```

teochenglim replied:
good idea. Changed.

SMillerDev commented on Formula/prometheus.rb:
```diff
@@ -22,12 +22,20 @@ def install
+      --storage.tsdb.path #{var}/prometheus' \
+      > #{etc}/prometheus.args"
+    config = "global:\n  scrape_interval: 15s\n\nscrape_configs:\n  - job_name: \"prometheus\"\n    static_configs:\n    - targets: [\"localhost:9090\"]\n"
+    system "echo '#{config}' > #{etc}/prometheus.yml"
```
Same here
```ruby
    (etc/"prometheus.yml").write <<~EOS
      global:
        scrape_interval: 15s
      scrape_configs:
        - job_name: "prometheus"
          static_configs:
          - targets: ["localhost:9090"]
    EOS
```

teochenglim replied:
good idea. Changed.

SMillerDev commented on Formula/prometheus.rb:
```diff
@@ -22,12 +22,20 @@ def install
 
-    Example configuration:
-      echo "--config.file ~/.config/prometheus.yml" > #{etc}/prometheus.args
+    To configure prometheus job, please edit your #{etc}/prometheus.yml file.
```
```suggestion
    Configuration for prometheus is located in the #{etc}/prometheus.yml file.
```

teochenglim replied:
good idea. Changed.

SMillerDev commented:
Hmm, I see promotheus als has a new version. Could you change the version to 2.14.0 and squash all your commits into one called `prometheus 2.14.0 and service fix`

teochenglim commented:
I would love to. But how do I change the sha?

SMillerDev commented:
It's the hash for the file it tries to download.

teochenglim commented:
* v2.14.0 seems need yarn as dependancy as package build from source.
* first download path is at bintray
https://homebrew.bintray.com/bottles/prometheus-2.14.0.catalina.bottle.tar.gz

```shell

$ brew install --build-from-source --debug --verbose Formula/prometheus.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FromPathLoader): loading /Users/teochenglim/homebrew-core/Formula/prometheus.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/go.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/prometheus.rb
/usr/bin/sandbox-exec -f /private/tmp/homebrew20191113-99044-gv16mv.sb nice ruby -W0 -I $LOAD_PATH -- /usr/local/Homebrew/Library/Homebrew/build.rb /Users/teochenglim/homebrew-core/Formula/prometheus.rb --verbose --debug
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FromPathLoader): loading /Users/teochenglim/homebrew-core/Formula/prometheus.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/go.rb
==> Downloading https://github.com/prometheus/prometheus/archive/v2.14.0.tar.gz
/usr/bin/curl -q --globoff --show-error --user-agent Homebrew/2.1.16-50-g5904aab\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 10.15.1\)\ curl/7.64.1 --location --silent --head --request GET https://github.com/prometheus/prometheus/archive/v2.14.0.tar.gz
Already downloaded: /Users/teochenglim/Library/Caches/Homebrew/downloads/4be74249400f52e78a84ed433eb4eea13faefacf1d52c569dbe4c56387273d06--prometheus-2.14.0.tar.gz
==> Verifying 4be74249400f52e78a84ed433eb4eea13faefacf1d52c569dbe4c56387273d06--prometheus-2.14.0.tar.gz checksum
tar xof /Users/teochenglim/Library/Caches/Homebrew/downloads/4be74249400f52e78a84ed433eb4eea13faefacf1d52c569dbe4c56387273d06--prometheus-2.14.0.tar.gz -C /private/tmp/d20191113-99045-8lbg5l
cp -pR /private/tmp/d20191113-99045-8lbg5l/prometheus-2.14.0/. /private/tmp/prometheus-20191113-99045-1mxjwjh/prometheus-2.14.0
chmod -Rf +w /private/tmp/d20191113-99045-8lbg5l
==> make build
cd web/ui/react-app && yarn --frozen-lockfile
/bin/sh: yarn: command not found
make: *** [web/ui/react-app/node_modules] Error 127
/usr/bin/curl --version
/usr/libexec/java_home --xml --failfast
/usr/libexec/java_home --xml --failfast
/usr/local/Homebrew/Library/Homebrew/debrew.rb:13:in `raise'
BuildError: Failed executing: make build
1. raise
2. ignore
3. backtrace
4. irb
5. shell
Choose an action: 1
==> Kept temporary files
Temporary files retained at /private/tmp/prometheus-20191113-99045-1mxjwjh

==> Formula
Path: /Users/teochenglim/homebrew-core/Formula/prometheus.rb
==> Configuration
HOMEBREW_VERSION: 2.1.16-50-g5904aab
ORIGIN: https://github.com/Homebrew/brew
HEAD: 5904aab98a32ab8ebbd74364e023cf9ed29ad795
Last commit: 11 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 982f0b56fa81f57591cef54011b80389114eaa4c
Core tap last commit: 9 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_DEV_CMD_RUN: 1
CPU: quad-core 64-bit kabylake
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
Clang: 11.0 build 1100
Git: 2.21.0 => /Library/Developer/CommandLineTools/usr/bin/git
/usr/bin/curl --version
Curl: 7.64.1 => /usr/bin/curl
/usr/libexec/java_home --xml --failfast
/usr/libexec/java_home --xml --failfast
Java: 12.0.1
macOS: 10.15.1-x86_64
CLT: 1100.0.33.8
Xcode: N/A
XQuartz: 2.7.11 => /opt/X11
==> ENV
HOMEBREW_CC: clang
HOMEBREW_CXX: clang++
MAKEFLAGS: -j4
CMAKE_PREFIX_PATH: /usr/local
CMAKE_INCLUDE_PATH: /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/libxml2:/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
CMAKE_LIBRARY_PATH: /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
PKG_CONFIG_LIBDIR: /usr/lib/pkgconfig:/usr/local/Homebrew/Library/Homebrew/os/mac/pkgconfig/10.15
HOMEBREW_GIT: git
HOMEBREW_SDKROOT: /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk
ACLOCAL_PATH: /usr/local/share/aclocal
PATH: /usr/local/Homebrew/Library/Homebrew/shims/mac/super:/usr/local/opt/go/bin:/usr/bin:/bin:/usr/sbin:/sbin

Error: prometheus 2.14.0 did not build
Logs:
     /Users/teochenglim/Library/Logs/Homebrew/prometheus/01.make
     /Users/teochenglim/Library/Logs/Homebrew/prometheus/00.options.out
Do not report this issue to Homebrew/brew or Homebrew/core!

/usr/bin/curl -q --globoff --show-error --user-agent Homebrew/2.1.16-50-g5904aab\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 10.15.1\)\ curl/7.64.1 --location https://api.github.com/search/issues\?q=prometheus\+state\%3Aopen\+repo\%3AHomebrew\%2Fhomebrew-core\+in\%3Atitle\&per_page=100 --header Accept:\ application/vnd.github.v3\+json --write-out '
'\%\{http_code\} --header Accept:\ application/vnd.github.antiope-preview\+json --dump-header /private/tmp/github_api_headers20191113-98166-1ke8bs8
These open issues may also help:
Fix prometheus service https://github.com/Homebrew/homebrew-core/pull/46618
node_exporter (for prometheus) doesn't report version information https://github.com/Homebrew/homebrew-core/issues/46212
prometheus not able to start with services mode! but manual working fine. https://github.com/Homebrew/homebrew-core/issues/44508

```

teochenglim commented:
in newer version "make build" including assets and need yarn. 
But then the "make common-build" do have the original go bin target

```shell
teochenglim@chenglim:/private/tmp/prometheus-20191113-99045-1mxjwjh/prometheus-2.14.0$ cat Makefile
# Copyright 2018 The Prometheus Authors
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

# Needs to be defined before including Makefile.common to auto-generate targets
DOCKER_ARCHS ?= amd64 armv7 arm64

REACT_APP_PATH = web/ui/react-app
REACT_APP_SOURCE_FILES = $(wildcard $(REACT_APP_PATH)/public/* $(REACT_APP_PATH)/src/* $(REACT_APP_PATH)/tsconfig.json)
REACT_APP_OUTPUT_DIR = web/ui/static/react
REACT_APP_NODE_MODULES_PATH = $(REACT_APP_PATH)/node_modules
REACT_APP_NPM_LICENSES_TARBALL = "npm_licenses.tar.bz2"

TSDB_PROJECT_DIR = "./tsdb"
TSDB_CLI_DIR="$(TSDB_PROJECT_DIR)/cmd/tsdb"
TSDB_BIN = "$(TSDB_CLI_DIR)/tsdb"
TSDB_BENCHMARK_NUM_METRICS ?= 1000
TSDB_BENCHMARK_DATASET ?= "$(TSDB_PROJECT_DIR)/testdata/20kseries.json"
TSDB_BENCHMARK_OUTPUT_DIR ?= "$(TSDB_CLI_DIR)/benchout"

include Makefile.common

DOCKER_IMAGE_NAME       ?= prometheus

$(REACT_APP_NODE_MODULES_PATH): $(REACT_APP_PATH)/package.json $(REACT_APP_PATH)/yarn.lock
	cd $(REACT_APP_PATH) && yarn --frozen-lockfile

$(REACT_APP_OUTPUT_DIR): $(REACT_APP_NODE_MODULES_PATH) $(REACT_APP_SOURCE_FILES)
	@echo ">> building React app"
	@./scripts/build_react_app.sh

.PHONY: assets
assets: $(REACT_APP_OUTPUT_DIR)
	@echo ">> writing assets"
	# Un-setting GOOS and GOARCH here because the generated Go code is always the same,
	# but the cached object code is incompatible between architectures and OSes (which
	# breaks cross-building for different combinations on CI in the same container).
	cd web/ui && GO111MODULE=$(GO111MODULE) GOOS= GOARCH= $(GO) generate -x -v $(GOOPTS)
	@$(GOFMT) -w ./web/ui

.PHONY: react-app-lint
react-app-lint:
	@echo ">> running React app linting"
	cd $(REACT_APP_PATH) && yarn lint:ci

.PHONY: react-app-lint-fix
react-app-lint-fix:
	@echo ">> running React app linting and fixing errors where possibe"
	cd $(REACT_APP_PATH) && yarn lint

.PHONY: react-app-test
react-app-test: | $(REACT_APP_NODE_MODULES_PATH) react-app-lint
	@echo ">> running React app tests"
	cd $(REACT_APP_PATH) && yarn test --no-watch --coverage

.PHONY: test
test: common-test react-app-test

.PHONY: npm_licenses
npm_licenses: $(REACT_APP_NODE_MODULES_PATH)
	@echo ">> bundling npm licenses"
	rm -f $(REACT_APP_NPM_LICENSES_TARBALL)
	find $(REACT_APP_NODE_MODULES_PATH) -iname "license*" | tar cfj $(REACT_APP_NPM_LICENSES_TARBALL) --transform 's/^/npm_licenses\//' --files-from=-

.PHONY: tarball
tarball: npm_licenses common-tarball

.PHONY: docker
docker: npm_licenses common-docker

.PHONY: build
build: assets common-build

.PHONY: build_tsdb
build_tsdb:
	GO111MODULE=$(GO111MODULE) $(GO) build -o $(TSDB_BIN) $(TSDB_CLI_DIR)

.PHONY: bench_tsdb
bench_tsdb: build_tsdb
	@echo ">> running benchmark, writing result to $(TSDB_BENCHMARK_OUTPUT_DIR)"
	@$(TSDB_BIN) bench write --metrics=$(TSDB_BENCHMARK_NUM_METRICS) --out=$(TSDB_BENCHMARK_OUTPUT_DIR) $(TSDB_BENCHMARK_DATASET)
	@$(GO) tool pprof -svg $(TSDB_BIN) $(TSDB_BENCHMARK_OUTPUT_DIR)/cpu.prof > $(TSDB_BENCHMARK_OUTPUT_DIR)/cpuprof.svg
	@$(GO) tool pprof --inuse_space -svg $(TSDB_BIN) $(TSDB_BENCHMARK_OUTPUT_DIR)/mem.prof > $(TSDB_BENCHMARK_OUTPUT_DIR)/memprof.inuse.svg
	@$(GO) tool pprof --alloc_space -svg $(TSDB_BIN) $(TSDB_BENCHMARK_OUTPUT_DIR)/mem.prof > $(TSDB_BENCHMARK_OUTPUT_DIR)/memprof.alloc.svg
	@$(GO) tool pprof -svg $(TSDB_BIN) $(TSDB_BENCHMARK_OUTPUT_DIR)/block.prof > $(TSDB_BENCHMARK_OUTPUT_DIR)/blockprof.svg
	@$(GO) tool pprof -svg $(TSDB_BIN) $(TSDB_BENCHMARK_OUTPUT_DIR)/mutex.prof > $(TSDB_BENCHMARK_OUTPUT_DIR)/mutexprof.svg
teochenglim@chenglim:/private/tmp/prometheus-20191113-99045-1mxjwjh/prometheus-2.14.0$ make build
cd web/ui/react-app && yarn --frozen-lockfile
yarn install v1.19.1
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[#########----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------] 69/1485^Cmake: *** [web/ui/react-app/node_modules] Interrupt: 2

teochenglim@chenglim:/private/tmp/prometheus-20191113-99045-1mxjwjh/prometheus-2.14.0$ make build
cd web/ui/react-app && yarn --frozen-lockfile
yarn install v1.19.1
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
✨  Done in 358.11s.
>> building React app
building React app
yarn run v1.19.1
$ react-scripts build
Creating an optimized production build...
Compiled successfully.

File sizes after gzip:

  299.07 KB  build/static/js/2.46ac6325.chunk.js
  23.12 KB   build/static/css/2.3f5e1097.chunk.css
  10.35 KB   build/static/js/main.d7eb4e3d.chunk.js
  1.06 KB    build/static/css/main.22aceb42.chunk.css
  769 B      build/static/js/runtime-main.56ad181a.js

The project was built assuming it is hosted at ./.
You can control this with the homepage field in your package.json.

The build folder is ready to be deployed.

Find out more about deployment here:

  https://bit.ly/CRA-deploy

✨  Done in 54.13s.
>> writing assets
# Un-setting GOOS and GOARCH here because the generated Go code is always the same,
# but the cached object code is incompatible between architectures and OSes (which
# breaks cross-building for different combinations on CI in the same container).
cd web/ui && GO111MODULE=on GOOS= GOARCH= go generate -x -v  -mod=vendor
doc.go
go run -mod=vendor assets_generate.go
writing assets_vfsdata.go
ui.go
curl -s -L https://github.com/prometheus/promu/releases/download/v0.5.0/promu-0.5.0.darwin-amd64.tar.gz | tar -xvzf - -C /var/folders/dp/qtmghhxn2d7d4xfcfk0qwl300000gp/T/tmp.Z5k53eXO
x promu-0.5.0.darwin-amd64/
x promu-0.5.0.darwin-amd64/promu
x promu-0.5.0.darwin-amd64/NOTICE
x promu-0.5.0.darwin-amd64/LICENSE
mkdir -p /Users/teochenglim/go/bin
cp /var/folders/dp/qtmghhxn2d7d4xfcfk0qwl300000gp/T/tmp.Z5k53eXO/promu-0.5.0.darwin-amd64/promu /Users/teochenglim/go/bin/promu
rm -r /var/folders/dp/qtmghhxn2d7d4xfcfk0qwl300000gp/T/tmp.Z5k53eXO
>> building binaries
GO111MODULE=on /Users/teochenglim/go/bin/promu build --prefix /private/tmp/prometheus-20191113-99045-1mxjwjh/prometheus-2.14.0
 >   prometheus
 >   promtool
 >   tsdb
teochenglim@chenglim:/private/tmp/prometheus-20191113-99045-1mxjwjh/prometheus-2.14.0$ make common-build
>> building binaries
GO111MODULE=on /Users/teochenglim/go/bin/promu build --prefix /private/tmp/prometheus-20191113-99045-1mxjwjh/prometheus-2.14.0
 >   prometheus
 >   promtool
 >   tsdb
$ ./prometheus --version
prometheus, version 2.14.0 (branch: non-git, revision: non-git)
  build user:       teochenglim@chenglim.teo
  build date:       20191113-12:02:53
  go version:       go1.13.4
```

teochenglim commented:
Changed from "make build" to "make common-build" and it is still failing, seems like the "make asset" is needed

```shell
$ brew install --build-from-source --debug --verbose Formula/prometheus.rb
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 1 tap (solo-io/tap).
==> Updated Formulae
solo-io/tap/glooctl ✔

/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FromPathLoader): loading /Users/teochenglim/homebrew-core/Formula/prometheus.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/go.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/prometheus.rb
/usr/bin/sandbox-exec -f /private/tmp/homebrew20191113-5855-fxf5mt.sb nice ruby -W0 -I $LOAD_PATH -- /usr/local/Homebrew/Library/Homebrew/build.rb /Users/teochenglim/homebrew-core/Formula/prometheus.rb --verbose --debug
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FromPathLoader): loading /Users/teochenglim/homebrew-core/Formula/prometheus.rb
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/go.rb
==> Downloading https://github.com/prometheus/prometheus/archive/v2.14.0.tar.gz
/usr/bin/curl -q --globoff --show-error --user-agent Homebrew/2.1.16-50-g5904aab\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 10.15.1\)\ curl/7.64.1 --location --silent --head --request GET https://github.com/prometheus/prometheus/archive/v2.14.0.tar.gz
Already downloaded: /Users/teochenglim/Library/Caches/Homebrew/downloads/4be74249400f52e78a84ed433eb4eea13faefacf1d52c569dbe4c56387273d06--prometheus-2.14.0.tar.gz
==> Verifying 4be74249400f52e78a84ed433eb4eea13faefacf1d52c569dbe4c56387273d06--prometheus-2.14.0.tar.gz checksum
tar xof /Users/teochenglim/Library/Caches/Homebrew/downloads/4be74249400f52e78a84ed433eb4eea13faefacf1d52c569dbe4c56387273d06--prometheus-2.14.0.tar.gz -C /private/tmp/d20191113-5856-lobypb
cp -pR /private/tmp/d20191113-5856-lobypb/prometheus-2.14.0/. /private/tmp/prometheus-20191113-5856-1i4rzof/prometheus-2.14.0
chmod -Rf +w /private/tmp/d20191113-5856-lobypb
==> make common-build
curl -s -L https://github.com/prometheus/promu/releases/download/v0.5.0/promu-0.5.0.darwin-amd64.tar.gz | tar -xvzf - -C /var/folders/dp/qtmghhxn2d7d4xfcfk0qwl300000gp/T/tmp.YKn4xDdn
x promu-0.5.0.darwin-amd64/
x promu-0.5.0.darwin-amd64/promu
x promu-0.5.0.darwin-amd64/NOTICE
x promu-0.5.0.darwin-amd64/LICENSE
mkdir -p /private/tmp/prometheus-20191113-5856-1i4rzof/prometheus-2.14.0/.brew_home/go/bin
cp /var/folders/dp/qtmghhxn2d7d4xfcfk0qwl300000gp/T/tmp.YKn4xDdn/promu-0.5.0.darwin-amd64/promu /private/tmp/prometheus-20191113-5856-1i4rzof/prometheus-2.14.0/.brew_home/go/bin/promu
rm -r /var/folders/dp/qtmghhxn2d7d4xfcfk0qwl300000gp/T/tmp.YKn4xDdn
>> building binaries
GO111MODULE=on /private/tmp/prometheus-20191113-5856-1i4rzof/prometheus-2.14.0/.brew_home/go/bin/promu build --prefix /private/tmp/prometheus-20191113-5856-1i4rzof/prometheus-2.14.0
 >   prometheus
# github.com/prometheus/prometheus/web
web/web.go:346:33: undefined: ui.Assets
web/web.go:360:14: undefined: ui.Assets
web/web.go:379:33: undefined: ui.Assets
web/web.go:1034:13: undefined: ui.Assets
!! command failed: build -o /private/tmp/prometheus-20191113-5856-1i4rzof/prometheus-2.14.0/prometheus -ldflags -X github.com/prometheus/common/version.Version=2.14.0 -X github.com/prometheus/common/version.Revision=non-git -X github.com/prometheus/common/version.Branch=non-git -X github.com/prometheus/common/version.BuildUser=teochenglim@chenglim.teo -X github.com/prometheus/common/version.BuildDate=20191113-12:11:20  -mod=vendor -a -tags netgo,builtinassets github.com/prometheus/prometheus/cmd/prometheus: exit status 2
make: *** [common-build] Error 1
/usr/bin/curl --version
/usr/libexec/java_home --xml --failfast
/usr/libexec/java_home --xml --failfast
/usr/local/Homebrew/Library/Homebrew/debrew.rb:13:in `raise'
BuildError: Failed executing: make common-build
1. raise
2. ignore
3. backtrace
4. irb
5. shell
Choose an action: 1
==> Kept temporary files
Temporary files retained at /private/tmp/prometheus-20191113-5856-1i4rzof

==> Formula
Path: /Users/teochenglim/homebrew-core/Formula/prometheus.rb
==> Configuration
HOMEBREW_VERSION: 2.1.16-50-g5904aab
ORIGIN: https://github.com/Homebrew/brew
HEAD: 5904aab98a32ab8ebbd74364e023cf9ed29ad795
Last commit: 12 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 982f0b56fa81f57591cef54011b80389114eaa4c
Core tap last commit: 9 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_DEV_CMD_RUN: 1
CPU: quad-core 64-bit kabylake
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
Clang: 11.0 build 1100
Git: 2.21.0 => /Library/Developer/CommandLineTools/usr/bin/git
/usr/bin/curl --version
Curl: 7.64.1 => /usr/bin/curl
/usr/libexec/java_home --xml --failfast
/usr/libexec/java_home --xml --failfast
Java: 12.0.1
macOS: 10.15.1-x86_64
CLT: 1100.0.33.8
Xcode: N/A
XQuartz: 2.7.11 => /opt/X11
==> ENV
HOMEBREW_CC: clang
HOMEBREW_CXX: clang++
MAKEFLAGS: -j4
CMAKE_PREFIX_PATH: /usr/local
CMAKE_INCLUDE_PATH: /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/libxml2:/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
CMAKE_LIBRARY_PATH: /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
PKG_CONFIG_LIBDIR: /usr/lib/pkgconfig:/usr/local/Homebrew/Library/Homebrew/os/mac/pkgconfig/10.15
HOMEBREW_GIT: git
HOMEBREW_SDKROOT: /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk
ACLOCAL_PATH: /usr/local/share/aclocal
PATH: /usr/local/Homebrew/Library/Homebrew/shims/mac/super:/usr/local/opt/go/bin:/usr/bin:/bin:/usr/sbin:/sbin

Error: prometheus 2.14.0 did not build
Logs:
     /Users/teochenglim/Library/Logs/Homebrew/prometheus/01.make
     /Users/teochenglim/Library/Logs/Homebrew/prometheus/00.options.out
Do not report this issue to Homebrew/brew or Homebrew/core!

/usr/bin/curl -q --globoff --show-error --user-agent Homebrew/2.1.16-50-g5904aab\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 10.15.1\)\ curl/7.64.1 --location https://api.github.com/search/issues\?q=prometheus\+state\%3Aopen\+repo\%3AHomebrew\%2Fhomebrew-core\+in\%3Atitle\&per_page=100 --header Accept:\ application/vnd.github.v3\+json --write-out '
'\%\{http_code\} --header Accept:\ application/vnd.github.antiope-preview\+json --dump-header /private/tmp/github_api_headers20191113-4656-6lmphm
These open issues may also help:
Fix prometheus service https://github.com/Homebrew/homebrew-core/pull/46618
node_exporter (for prometheus) doesn't report version information https://github.com/Homebrew/homebrew-core/issues/46212
prometheus not able to start with services mode! but manual working fine. https://github.com/Homebrew/homebrew-core/issues/44508

```

teochenglim commented:
@SMillerDev newer version has some changes and require "yarn" and their build script stop because it is using "/bin/sh" and can't find "yarn". 

I am not able to just change the binary and sha and it will work. details log above. I have change it to this 64 bits binary 
https://github.com/prometheus/prometheus/releases/download/v2.14.0/prometheus-2.14.0.darwin-amd64.tar.gz

Everything work as expected but doesn't have "build from source" anymore.

square all commits into 1 now.

SMillerDev commented:
leave the version bump then, homebrew formula should only build from source.

teochenglim commented:
Reverted back to v2.13.1

teochenglim commented:
@SMillerDev ready to merge?

SMillerDev commented:
Thanks a lot @teochenglim

chenrui333 commented:
Thanks @teochenglim!!

teochenglim commented:
Any mistake i made here? Why is it closed and not merge?
